src/overlays/actors/ovl_Oceff_Wipe3/z_oceff_wipe3.c

```diff
@@ -29,12 +28,82 @@ const ActorInit Oceff_Wipe3_InitVars = {
     (ActorFunc)OceffWipe3_Draw,
 };
 
-#endif
+#include "assets/overlays/ovl_Oceff_Wipe3/ovl_Oceff_Wipe3.c"
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Wipe3/OceffWipe3_Init.s")
+s32 D_80989130;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Wipe3/OceffWipe3_Destroy.s")
+void OceffWipe3_Init(Actor* thisx, PlayState* play) {
+    OceffWipe3* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Wipe3/OceffWipe3_Update.s")
+    Actor_SetScale(&this->actor, 0.1f);
+    this->counter = 0;
+    this->actor.world.pos = GET_ACTIVE_CAM(play)->eye;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Wipe3/OceffWipe3_Draw.s")
+void OceffWipe3_Destroy(Actor* thisx, PlayState* play) {
+    OceffWipe3* this = THIS;
+
+    func_80115D5C(&play->state);
+    play->msgCtx.unk120B0 = 0;
+}
+
+void OceffWipe3_Update(Actor* thisx, PlayState* play) {
+    OceffWipe3* this = THIS;
+
+    this->actor.world.pos = GET_ACTIVE_CAM(play)->eye;
+    if (this->counter < 100) {
+        this->counter++;
+    } else {
+        Actor_MarkForDeath(&this->actor);
+    }
+}
+
+void OceffWipe3_Draw(Actor* thisx, PlayState* play) {
+    u32 scroll = play->state.frames & 0xFFF;
+    OceffWipe3* this = THIS;
+    f32 z;
+    u8 alpha;
+    s32 pad[2];
+    Vec3f eye = GET_ACTIVE_CAM(play)->eye;
+    Vtx* vtxPtr;
+    Vec3f quakeOffset;
+
+    Camera_GetQuakeOffset(&quakeOffset, GET_ACTIVE_CAM(play));
+
+    vtxPtr = sSariaSongFrustumVtx;
+
+    if (this->counter < 32) {
+        z = Math_SinS(this->counter << 9) * 1220.0f;
```

Comment:
```suggestion
        z = Math_SinS(this->counter * 512) * 1220.0f;
```
I think this would make more sense if it matches